tv_apps/smart-system/js/app_modal_dialog.js

```diff
@@ -385,20 +431,23 @@
           checkbox.parentNode.classList.add('hidden');
         }
 
-        focusManager.focus();
         break;
     }
 
     this.app.browser.element.setAttribute('aria-hidden', true);
-    this.element.classList.add('visible');
+    this.element.focus();
+    if (this.element.classList.contains('closed')) {
+      this.element.open();
+    }
   };
 
   AppModalDialog.prototype.hide = function amd_hide() {
     this.simpleKeyNavigation.blur();
     this.simpleKeyNavigation.stop();
     this.app.browser.element.removeAttribute('aria-hidden');
-    this.element.classList.remove('visible');
-    focusManager.focus();
+    this.element.focus();
```

Comment:
Please add comments here to explain that we should focus to container while playing animation.